x-pack/plugins/ml/public/application/contexts/kibana/use_cases_modal.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback } from 'react';
+import { stringHash } from '@kbn/ml-string-hash';
+import { CommentType } from '@kbn/cases-plugin/common';
+import { useMlKibana } from './kibana_context';
+import type { MappedEmbeddableTypeOf, MlEmbeddableTypes } from '../../../embeddables';
+
+/**
+ * Returns a callback for opening the cases modal with provided attachment state.
+ */
+export const useCasesModal = <EmbeddableType extends MlEmbeddableTypes>(
+  embeddableType: EmbeddableType
+) => {
+  const {
+    services: { cases },
+  } = useMlKibana();
+
+  const selectCaseModal = cases?.hooks.getUseCasesAddToExistingCaseModal();
+
+  return useCallback(
+    (persistableState: Partial<Omit<MappedEmbeddableTypeOf<EmbeddableType>, 'id'>>) => {
+      const persistableStateAttachmentState = {
+        ...persistableState,
+        // Creates unique id based on the input
+        id: stringHash(JSON.stringify(persistableState)).toString(),
+      };
+
+      selectCaseModal!.open({
```

Comment:
What happens when selectCaseModal is undefined? Is it possible that it will be undefined if `cases` for some reasons is unavailable?

Reply:
The parent component checks the cases plugin's availability, but I added an explicit check [here](https://github.com/elastic/kibana/pull/139628/commits/bbb0f523627547de933a699827129045370b4787) to throw an error. Just to better indicate the issue during development in the future.